tests/cpp/test_multidevice_sharding.cpp

```diff
@@ -147,20 +143,21 @@ TEST_P(ShardingTest, UnshardedGlobalInput) {
 // Test memory allocation of multidevice fusion with sharded input
 // and replicated intermediates and output.
 TEST_P(ShardingTest, ShardGlobalInput) {
-  auto concreteTv = GetParam();
-  int sharded_dim = 0;
+  auto [concreteTv, sharded_dim] = GetParam();
   std::unique_ptr<Fusion> fusion = std::make_unique<Fusion>();
   FusionGuard fg(fusion.get());
   int num_devices = communicator->size();
   std::vector<int64_t> devices(num_devices);
   std::iota(devices.begin(), devices.end(), 0);
   DeviceMesh mesh(devices);
-  std::vector<int64_t> unsharded_input_size = {num_devices, 3, 2};
+  std::vector<int64_t> unsharded_input_size = {3, 2, 5};
+  unsharded_input_size[sharded_dim] = num_devices;
 
-  TensorView* tv0 = concreteTv ? makeConcreteTensor(unsharded_input_size)
-                               : makeContigTensor(3);
+  TensorView* tv0 = concreteTv
+      ? makeConcreteTensor(unsharded_input_size)
+      : makeSymbolicTensor(unsharded_input_size.size());
   TensorView* tv1 = set(tv0);
-  TensorView* tv2 = add(tv1, tv1);
+  TensorView* tv2 = sum(tv1, {1});
```

Comment:
This seems to be a behavior change not a cleanup as the title suggested. Is there a reason behind this?

Reply:
(The same comment goes to line 175)

Reply:
yeah, these are changes from some random testing - will revert back.